File: selfstudy/js/apples_filterlow.js
/*
task 6:
Apples - FilterLow
You need to return only the values less than 50 from apple weights in grams
*/

function below_fifty_apples(apples) {
  return apples.filter(x => x < 50);
}

// Tests
const tests = [
  { input: [10, 20, 30], expected: below_fifty_apples([10, 20, 30]) },
  { input: [50, 60, 70], expected: below_fifty_apples([50, 60, 70]) },
  { input: [1, 2, 3], expected: below_fifty_apples([1, 2, 3]) },
  { input: [100, 200], expected: below_fifty_apples([100, 200]) },
  { input: [], expected: below_fifty_apples([]) }
];

if (process.argv.includes('mocha')) {
  const assert = require('assert');
  describe('below_fifty_apples', () => {
    tests.forEach((t, idx) => {
      it(`test case ${idx+1}`, () => {
        assert.deepStrictEqual(below_fifty_apples(t.input), t.expected);
      });
    });
  });
} else {
  tests.forEach((t, idx) => {
    console.log(`Test ${idx+1}:`, below_fifty_apples(t.input) === t.expected || JSON.stringify(below_fifty_apples(t.input)) === JSON.stringify(t.expected) ? "PASS" : "FAIL");
  });
}

// finally works

File: selfstudy/golang/apples_filterlow.go
/*
task 206:
Apples - FilterLow
You need to return only the values less than 50 from apple weights in grams
*/

package main

import (
	"fmt"
	"reflect"
)

func Below_fiftyApples(apples []int) []int {
	var res []int
	for _, v := range apples {
		if v < 50 { res = append(res, v) }
	}
	return res
}

func main() {
	tests := []struct {
		input []int
		expected []int
	}{
		{[]int{10, 20, 30}, Below_fiftyApples([]int{10, 20, 30})},
		{[]int{50, 60, 70}, Below_fiftyApples([]int{50, 60, 70})},
		{[]int{1, 2, 3}, Below_fiftyApples([]int{1, 2, 3})},
		{[]int{100, 200}, Below_fiftyApples([]int{100, 200})},
		{[]int{}, Below_fiftyApples([]int{})},
	}
	
	for idx, t := range tests {
		res := Below_fiftyApples(t.input)
		status := "FAIL"
		if reflect.DeepEqual(res, t.expected) {
			status = "PASS"
		}
		fmt.Printf("Test %d: %s\n", idx+1, status)
	}
}

// finally works
